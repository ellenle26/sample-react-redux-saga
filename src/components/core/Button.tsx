import classNames from "classnames";

const Button: React.FC<ButtonProps> = (props) => {
    const {text, children, type, className, onClick} = props;
    
    return <button className={classNames(className, 
        "py-5 px-2 w-40 text-sm font-normal border border-solid hover:opacity-80", {
        "bg-gray_": type === "contained"
    })} onClick = {event => onClick && onClick(event)}>{!text ? children: text}</button>
}

interface ButtonProps {
    className?: string;
    text?: string;
    children?: any;
    type?: "outline" | "contained";
    onClick?: (event: any) => void;
}
export default Button;