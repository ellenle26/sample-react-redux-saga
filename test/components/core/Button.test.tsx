/**
 * @jest-environment jsdom
*/

import Button from "@components/core/Button";
import {fireEvent, render} from "@testing-library/react";
import React from "react";

test("snapshot testing", () => {
  const renderResult = render(<Button />);
  expect(renderResult.asFragment()).toMatchSnapshot();
});

test("having text testing", () => {
    const renderResult = render(<Button text="button" />);
    expect(renderResult.container.innerHTML).toMatch("button");
});

test("click testing", () => {
    const handleClick = jest.fn();
    const renderResult = render(<Button onClick={handleClick} />);
    fireEvent.click(renderResult.container.getElementsByTagName("button")[0]);
    expect(handleClick).toHaveBeenCalledTimes(1);
});