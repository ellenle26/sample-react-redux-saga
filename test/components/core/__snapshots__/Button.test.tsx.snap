// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<DocumentFragment>
  <button
    class="py-5 px-2 w-40 text-sm font-normal border border-solid hover:opacity-80"
  />
</DocumentFragment>
`;
